tools/tcpaccept.py

```diff
@@ -147,6 +155,51 @@
 }
 """
 
+bpf_text_tracepoint = """
+TRACEPOINT_PROBE(sock, inet_sock_set_state)
+{
+    if (args->protocol != IPPROTO_TCP)
+        return 0;
+    u32 pid = bpf_get_current_pid_tgid();
+    struct sock *newsk = (struct sock *)args->skaddr;
+
+    // pull in details
+    u16 family = 0, lport = 0;
+    family = newsk->__sk_common.skc_family;
+    lport = newsk->__sk_common.skc_num;
```

Comment:
The tracepoint:
```
          TP_STRUCT__entry(
                __field(const void *, skaddr)
                __field(int, oldstate)
                __field(int, newstate)
                __field(__u16, sport)
                __field(__u16, dport)
                __field(__u16, family)
                __field(__u8, protocol)
                __array(__u8, saddr, 4)
                __array(__u8, daddr, 4)
                __array(__u8, saddr_v6, 16)
                __array(__u8, daddr_v6, 16)
        ),
```
You can get family etc directly from tracepoint, why you still use newsk?

Reply:
I just think for modifying least, 
and not sure using args->skaddr->__sk_common.skc_family to get the family will more overhead than args->family.